.gitignore

```diff
@@ -119,3 +119,9 @@ compile_commands.json
 
 # Domain Proxy
 dp/cloud/helm/dp/charts/domain-proxy/certificates/**/*
+
+# clangd build files
+.clangd/
```

Comment:
We might want to combine C related stuff? E.g. with [L22 f](https://github.com/magma/magma/pull/12342/files#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947R22-R23).